llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -2445,6 +2445,11 @@ bool ModuleAddressSanitizer::InstrumentGlobals(IRBuilder<> &IRB, Module &M,
     // zero so we can copy the metadata over as is.
     NewGlobal->copyMetadata(G, 0);
 
+    // Set sanitizer metadata for newly created global,
+    // if it doesn't have it.
```

Comment:
Should we comment here that this may be a workaround for metadata lost between front end and this instrumentation pass?